system/core/pages.php

```diff
@@ -133,7 +133,7 @@ function loadPage(RequestContext $O_O) {
 			}
 			$_node = hypha_getPage($O_O->getContentLanguage(), $_name);
 
-			if ($_node) {
+			if ($_node && (!in_array($_node->getAttribute('private'), ['true', '1', 'on']) || isUser())) {
```

Comment:
Doesn't this mean the user gets a 404 when they're not logged in, rather than a message saying they should login? This is better from a strict security perspective, but leakage of private page names does not seem like a big issue here and could be less important than usability?

Reply:
When, when a not-logged-in user tries the access a private page they will see a login-to-view message.

Reply:
I've added a bit more to the commit message for this change:

>    Additionally, the check for a private page now happens in loadPage,       
>    rather than in the page's show(). This helps to make sure that the flag               
>    is always checked, and makes sure a 404 is returned.